Add usage reporter to track feature flags (#1661)
grafana/agent: docs/user/configuration/flags.md

* `dynamic-config`: Enable support for [dynamic configuration]({{< relref "./dynamic-config" >}})
* `extra-scrape-metrics`: When enabled, additional time series  are exposed for each metrics instance scrape. See [Extra scrape metrics](https://prometheus.io/docs/prometheus/latest/feature_flags/#extra-scrape-metrics).

> ### Report usage of feature flags

Reviewer: ```suggestion
### Report use of feature flags
```

> Grafana Agent provides the flag `-usage-report` to report the usage of feature flags to grafana.com.

Reviewer: ```suggestion
Grafana Agent provides the flag `-usage-report` to report the use of feature flags to grafana.com.
```

> This helps to understand which enabled features are being used of your running Agent instance (all data sent is anonymous).

Reviewer: ```suggestion
This helps to understand which enabled features are being used as part of your running Agent instance. All data data sent is anonymous.
```

* `dynamic-config`: Enable support for [dynamic configuration]({{< relref "./dynamic-config" >}})
* `extra-scrape-metrics`: When enabled, additional time series  are exposed for each metrics instance scrape. See [Extra scrape metrics](https://prometheus.io/docs/prometheus/latest/feature_flags/#extra-scrape-metrics).

### Report use of feature flags

> Grafana Agent provides the flag `-usage-report` to report the use of feature flags to grafana.com.

Reviewer 2: I think we should re-consider the flag name since we're going by opt-out. How about `-disable-usage-reports`, or `-disable-reporting`?

> This helps to understand which enabled features are being used as part of your running Agent instance.

Reviewer: ```suggestion
This helps you understand which enabled features are being used as part of your running Grafana Agent instance.
```
